### PR TITLE
Move reusable code from `sparsegpt_utils` to common and hf utils

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
`sparsegpt_utils` has some mappings and helper functions that are useful in other parts of Olive too. This PR refactors it and puts the reusable code into `olive.common.utils` and `olive.model.hf_utils`. 

Also cleaned up the imports in `hf_utils` by moving `transformers` imports to top of file. `transformers` is an olive requirement so we don't need to lazy import it unless it is a version specific import. 

## Checklist before requesting a review
- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/common/utils.py [hunk 1]
```diff
@@ -162,3 +162,28 @@ def tensor_data_to_device(data, device: str):
...
+        # return module if attr is empty
+        return module
+
+    attr = attr.split(".")
+    for a in attr:
+        try:
+            module = getattr(module, a)
+        except AttributeError:
+            if fail_on_not_found:
+                raise AttributeError(f"Attribute {attr} not found.")
```
Comment (guotuofeng) [inline olive/common/utils.py:186]: I suppose raise is enough. If we want to re-raise the exception, we could use raise from e

Comment (jambayk) [reply olive/common/utils.py:186]: The error message is different from the original one. The original message only mentions `a` but we want to raise with `attr` in the message.

Comment (guotuofeng) [reply olive/common/utils.py:186]: how about [https://stackoverflow.com/questions/24752395/python-raise-from-usage](https://stackoverflow.com/questions/24752395/python-raise-from-usage)?

Comment (guotuofeng) [reply olive/common/utils.py:186]: [https://github.com/microsoft/OLive/blob/9028fd5dd6def368ce93dda14140fe01fff8b3ed/olive/passes/onnx/quantization.py#L412](https://github.com/microsoft/OLive/blob/9028fd5dd6def368ce93dda14140fe01fff8b3ed/olive/passes/onnx/quantization.py#L412)

Comment (jambayk) [reply olive/common/utils.py]: Thanks! just updated it. 
The `raise from` method is much better since it returns a better trace.

**Review by guotuofeng — APPROVED** [review]

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**